pkg/lightning/backend/tidb/tidb.go

```diff
@@ -712,11 +769,17 @@ func (be *tidbBackend) WriteRowsToDB(ctx context.Context, tableName string, colu
 	}
 	is := insertStmt.String()
 	stmtTasks := make([]stmtTask, 0, len(rows))
+	var values []any
 	for _, row := range rows {
 		var finalInsertStmt strings.Builder
 		finalInsertStmt.WriteString(is)
-		finalInsertStmt.WriteString(row.insertStmt)
-		stmtTasks = append(stmtTasks, stmtTask{[]tidbRow{row}, finalInsertStmt.String()})
+		if be.stmtCache != nil {
+			finalInsertStmt.WriteString(row.preparedInsertStmt)
+			values = append(values, row.values...)
```

Comment:
please add ut to cover the case that Batch failed and fallback to single row insert with prepared-stmt